scripts/install

```diff
@@ -181,13 +194,26 @@ exec_occ ldap:test-config \'\' \
 # MOUNT HOME FOLDERS AS EXTERNAL STORAGE
 #=================================================
 
+# Define a function to add an external storage
+# Create the external storage for the given folders and enable sharing
+create_external_storage() {
```

Comment:
I'm still a bit puzzled about the balance between:
* this duplicated function (deleted from `_common.sh`), which is quite atomic and well defined as such
* experimental helpers, which have the "right" to be factorized and hence defined in `_common.sh`...

The only important difference I see is that this function is specific to this application (Nextcloud), whereas experimental helpers are designed to be versatile.

Is that the general rule in your mind, @maniackcrudelis ?

I thought about this when planning to apply the latest template to discourse package, and stumbled upon these [internal helpers](https://github.com/YunoHost-Apps/discourse_ynh/blob/master/scripts/_common.sh#L30-L69). I so understand that I need to publish them as experimental helpers to be able to factorize my code...? :wink:

Cc @YunoHost-Apps/apps-group

Reply:
That's indeed the idea, to factorize only parts of the code that's will be reused by every packagers. So we won't have to read the code to know what it's going on in it.

In opposition with specific functions for which you don't know what's into it.
Considering also that Experimental helpers are supposed to be official by the time.

Here I indeed defined a function into the install script to not rewrite the same thing many times. But the function is where we're going to to use it. Not elsewhere, so the code stays easily readable.